packages/web3-wrapper/src/web3_wrapper.ts

```diff
@@ -489,6 +495,21 @@ export class Web3Wrapper {
     public async setHeadAsync(blockNumber: number): Promise<void> {
         await this._sendRawPayloadAsync<void>({ method: 'debug_setHead', params: [this._web3.toHex(blockNumber)] });
     }
+    /**
+     * Returns either NodeType.Geth or NodeType.Ganache depending on the type of
+     * the backing Ethereum node. Throws for any other type of node. This
+     * function caches the result and so subsequent calls are fast.
+     */
+    public async getNodeTypeAsync(): Promise<NodeType> {
```

Comment:
Since both the assertions and __blockchain_lifecycle.ts__ switch on the `NodeType`, I decided to add the common functionality here.